pkg/internal/apis/certmanager/v1alpha2/conversion.go

```diff
@@ -126,3 +126,7 @@ func Convert_certmanager_CertificateRequestSpec_To_v1alpha2_CertificateRequestSp
 	out.CSRPEM = in.Request
 	return nil
 }
+
+func Convert_certmanager_CAIssuer_To_v1alpha2_CAIssuer(in *certmanager.CAIssuer, out *v1alpha2.CAIssuer, s conversion.Scope) error {
```

Comment:
I think you might need to add a line here (and in other conversion functions) to create an empty list rather than nil when converting from versions without the OCSP field.
See : https://github.com/jetstack/cert-manager/blob/11353f2936c42f3b2bc91d62be969bf7efa9f50e/pkg/internal/apis/certmanager/v1/conversion.go#L29-L33 for example

That's what causes the bazel roundtrip tests to fail, I think.

Reply:
The `OCSPServer` field doesn't exists in the other versions. In your example it is about a conversion of existing fields.

The round tripping test complains that the OCSPServer is empty:
```
round tripping to cert-manager.io/v1alpha3, Kind=ClusterIssuer v1alpha3.ClusterIssuer
...
          			CA: &certmanager.CAIssuer{
          				SecretName:            ".Ǐɽ逢ûȣ馣喀Ʃň{漯ʭF柱*gƋ",
        - 				CRLDistributionPoints: []string{},
        + 				CRLDistributionPoints: nil,
        - 				OCSPServer:            []string{"ƭI螞樆"},
        + 				OCSPServer:            nil,
          			},
```

Reply:
It should exist in other versions too otherwise you will end up losing data

Reply:
Tests are passing.

The `Certificate` struct in the x509 package defines the `OCSPServer` field as `[]string`. It is like the `CRLDistributionPoints` field.  Also the RFC doesn't specify that the value must be an URL.